docs/modules/ROOT/pages/join-two-streams.adoc

```diff
@@ -0,0 +1,343 @@
+```
+
+== Step 6. Create a Materialized View
+
+In this step, you'll define a job to run this streaming query in the background and store the results in a materialized view, using a Hazelcast map.
+
+. Create a mapping to a Hazelcast map called `orders_shipped_within_7_days`.
++
+```sql
+CREATE OR REPLACE MAPPING orders_shipped_within_7_days(
+  __key INT, <1>
+  order_ts TIMESTAMP WITH TIME ZONE, <2>
+  total_amount DOUBLE,
+  customer_name VARCHAR,
+  shipment_id VARCHAR,
+  ship_ts TIMESTAMP WITH TIME ZONE,
+  warehouse VARCHAR
+)
+TYPE IMAP
+  OPTIONS (
+    'keyFormat' = 'int', <1>
+    'valueFormat' = 'json-flat'); <2>
+```
++
+<1> The first column must be named `__key`. This column is mapped to the key of map entries.
+<2> The other columns must appear in the same order as the streaming query results so that the data types are mapped correctly.
+
+. Create the job.
++
+```sql
+CREATE JOB get_orders_shipped_within_7_days AS
+  SINK INTO orders_shipped_within_7_days <1>
+  SELECT o.id AS __key, <2>
+    o.order_ts,
+    o.total_amount,
+    o.customer_name,
+    s.id AS shipment_id,
+    s.ship_ts,
+    s.warehouse
+  FROM orders_ordered o JOIN shipments_ordered s <1>
+  ON o.id = s.order_id AND s.ship_ts BETWEEN o.order_ts AND o.order_ts + INTERVAL '7' DAYS;
+```
++
+<1> Insert the results into the `orders_shipped_within_7_days ` map.
+<2> Make sure that the selected fields are in the same order as the mapping to the `orders_shipped_within_7_days ` map.
+
+. Query the map to make sure that the job is working.
++
+```sql
+SELECT * FROM orders_shipped_within_7_days;
+```
+
+You should see the following:
+
+```
++------------+-------------------------+-------------------------+--------------------+--------------------+-------------------------+--------------------+
+|       __key|order_ts                 |             total_amount|customer_name       |shipment_id         |ship_ts                  |warehouse           |
++------------+-------------------------+-------------------------+--------------------+--------------------+-------------------------+--------------------+
+|           2|2022-03-29T17:02:20Z     |                164839.31|Alex                |ship-xf72808        |2022-03-31T02:04:13Z     |UPS                 |
+|           1|2022-03-29T06:01:18Z     |                133548.84|Amal                |ship-ch83360        |2022-03-31T18:13:39Z     |UPS                 |
+|           3|2022-03-29T13:44:10Z     |                 90427.66|Hao                 |ship-kr47454        |2022-03-31T20:47:09Z     |DHL                 |
++------------+-------------------------+-------------------------+--------------------+--------------------+-------------------------+--------------------+
+```
+
+If you left this query running, it would continue to add new results for orders shipped within 7 days. You can connect your applications to the Hazelcast cluster and query this map to get further insights.
```

Comment:
Do we have a resource regarding managing streaming queries? (cancel, suspend, resume..)? It would be good to link here or below in next steps.

Reply:
yes, I'll add it 👍